airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -2109,3 +2128,265 @@ def execute(self, context: Any):
     def on_kill(self):
         if self.job_id and self.cancel_on_kill:
             self.hook.cancel_job(job_id=self.job_id, project_id=self.project_id, location=self.location)
+
+
+class GreatExpectationsBigQueryOperator(BaseOperator):
```

Comment:
I'm not much of a committer so take this with a grain of salt as a user suggestion, but:
I'd like to see this broken into two classes, `GreatExpectationsBaseOperator`, and `GreatExpectationsBigQueryOperator`.

The former could have a method `_configure_project(…)` composed of calls to `_configure_datasouces(…)` `_configure_stores(…)` and `_configure_docs(…)` which would effectively throw a not implemented exception. Its init would not mention the gcp specific params but still cover the validations, email, and other parameters that are general. The latter could then implement these methods to build the config used in the base's execute method. Its init could call super with kwargs. This would leave it open to users if they wanted to run expectations on one of the other supported datasources to implement something that could.

I'm a little unsure of how you could support EG a user that wants to DQ check a mysql db and have the stores and docs in gcp vs checking a mysql db with the stores and docs on aws. I almost want to recommend mixins, but am unsure.

Reply:
All for it.

Reply:
So this, sounds like an answer to https://github.com/apache/airflow/pull/11113#discussion_r494165469 - I would be in favor of having generic GE. I'm not familiar with this tool but from https://github.com/apache/airflow/pull/11113#discussion_r494205186 it look like we should be able to create one

Reply:
Ok, so we'll work on creating a `GreatExpectationsBaseOperator` and extending it using the `GreatExpectationsBigQueryOperator`.